habitat/sims/rearrangement/rearrangement_simulator.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python3
+
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import numpy as np
+
+from habitat.core.registry import registry
+from habitat.core.simulator import Config
+from habitat.sims.habitat_simulator.habitat_simulator import HabitatSim
+from habitat_sim.agent import ActionSpec, ActuationSpec
+from habitat_sim.nav import NavMeshSettings
+from habitat_sim.physics import MotionType
+from habitat_sim.utils.common import quat_from_coeffs, quat_to_magnum
+
+
+@registry.register_simulator(name="RearrangementSim-v0")
+class RearrangementSim(HabitatSim):
```

Comment:
This all should be broken into new members of `HabitatSim` + `RearrangementTask` changes.